From 7db664fcb5a4d34aea7b005ae7230d3d5b1485c0 Mon Sep 17 00:00:00 2001
From: Jooyeon Lee <jylee256.lee@lge.com>
Date: Tue, 11 Jul 2023 14:06:32 +0900
Subject: [PATCH] Pass the RNN nodes not checking if it iss half_pixel_center
 or not

:Release Notes:
Pass the RNN nodes not checking if it iss half_pixel_center or not

:Detailed Notes:
Pass the RNN nodes not checking if it iss half_pixel_center or not

:Testing Performed:
Build results and WRP-20342.

:QA Notes:

:Issues Addressed:
[WRP-21178] Verification of RTK nnapi about caching, latency, etc.
---
Upstream-Status: Pending

 tensorflow/lite/delegates/nnapi/nnapi_delegate.cc | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/tensorflow/lite/delegates/nnapi/nnapi_delegate.cc b/tensorflow/lite/delegates/nnapi/nnapi_delegate.cc
index 06cd5d34b65..590d994559c 100644
--- a/tensorflow/lite/delegates/nnapi/nnapi_delegate.cc
+++ b/tensorflow/lite/delegates/nnapi/nnapi_delegate.cc
@@ -2615,9 +2615,11 @@ bool NNAPIDelegateKernel::Validate(
         Expect(!builtin->align_corners,
                NNAPIValidationFailureType::kUnsupportedOperandValue,
                "NNAPI does not support align_corners == true.", &val_ctx);
-        Expect(!builtin->half_pixel_centers,
-               NNAPIValidationFailureType::kUnsupportedOperandValue,
-               "NNAPI does not support half_pixel_centers == true.", &val_ctx);
+        //Expect(!builtin->half_pixel_centers,  /* In webOS, just go through it not checking if half_pixel_centers */
+               //NNAPIValidationFailureType::kUnsupportedOperandValue,
+               //"NNAPI does not support half_pixel_centers == true.", &val_ctx);
+        TFLITE_LOG_PROD_ONCE(tflite::TFLITE_LOG_INFO, "builtin->half_pixel_centers:%d, but go through it",
+                builtin->half_pixel_centers);
       }
     } break;
     case kTfLiteBuiltinSqueeze: {
